src/main/java/com/sandvoxel/generitech/common/tileentities/power/TestPower.java

```diff
@@ -26,195 +27,44 @@
-                //System.out.println(abbs);
-                if (abbs > 100) {
-                    if (!worldObj.isRemote) {
-                        GeneriTech.network.sendToServer(new PacketPower(T0transfer, pos.getX(), pos.getZ(), pos.getY() + -1));
-
-                    }
-                    this.container.takePower(T0transfer, false);
-                } else {
-                    if (te2 instanceof ITeslaConsumer && abbs != 0 && abbs / 2 != 0) {
-                        if (!worldObj.isRemote) {
-                            GeneriTech.network.sendToServer(new PacketPower(abbs / 2, pos.getX(), pos.getZ(), pos.getY() + -1));
-                        }
-                        this.container.takePower(abbs / 2, false);
-                        if (te2 instanceof ITeslaConsumer && abbs == 1) {
-                            if (!worldObj.isRemote) {
-                                GeneriTech.network.sendToServer(new PacketPower(1, pos.getX(), pos.getZ(), pos.getY() + -1));
-                            }
-                            this.container.takePower(1, false);
-                        }
-                    }
-                }
-
-            }
+            TeslaUtils.distributePowerToAllFaces(wold,pos,100,false);
 
         }
 
 
-        flag[0] = this.canConnectTo(worldIn, pos.up());
-        flag[1] = this.canConnectTo(worldIn, pos.down());
-        flag[2] = this.canConnectTo(worldIn, pos.north());
-        flag[3] = this.canConnectTo(worldIn, pos.south());
-        flag[4] = this.canConnectTo(worldIn, pos.east());
-        flag[5] = this.canConnectTo(worldIn, pos.west());
-
 
-    }
 
 
-    public long getTransfer(BlockPos pos) {
-        TileEntity tile = worldObj.getTileEntity(pos);
-
-        if (tile instanceof ITeslaHolder) {
-            return ((ITeslaHolder) tile).getCapacity() - ((ITeslaHolder) tile).getStoredPower();
-        }
-
-        return 0;
+    @Override
+    public IInventory getInternalInventory() {
+        return inventory;
     }
+    @Override
+    public void onChangeInventory(IInventory inv, int slot, InventoryOperation operation, ItemStack removed, ItemStack added) {
 
-
-    public long getLeft() {
-        BlockPos pos = getPos();
-        TileEntity tile = worldObj.getTileEntity(pos.up());
-
-        if (tile instanceof ITeslaHolder) {
-            return ((ITeslaHolder) tile).getStoredPower();
-        }
-
-        return 0;
     }
 
-
-    public boolean canConnectTo(IBlockAccess worldIn, BlockPos pos) {
-        Block block = worldIn.getBlockState(pos).getBlock();
-        if (block == Blocks.BLOCK_FURNACE.getBlock() || block == Blocks.BLOCK_PULVERIZER.getBlock() || block == Blocks.BLOCK_CABLES.getBlock()) {
-            return true;
-        } else {
-            return false;
-        }
+    @Override
+    public int[] getAccessibleSlotsBySide(EnumFacing side) {
+        return new int[0];
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS134)